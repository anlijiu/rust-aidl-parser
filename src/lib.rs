mod ast;
mod error;
pub mod types;

pub fn parse(inputs: &[&str]) -> Result<Vec<types::File>, error::ParseError> {
    let files = inputs.iter().map(|i| {
        let lookup = line_col::LineColLookup::new(i);
        ast::rules::file(i, &lookup).map_err(error::ParseError::Peg)
    });

    files.collect()
}
