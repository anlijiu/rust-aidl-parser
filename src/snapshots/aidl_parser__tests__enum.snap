---
source: src/lib.rs
expression: res

---
Enum(
  name: "Paprika",
  elements: [
    EnumElement(
      name: "ELEMENT1",
      value: Some("3"),
      doc: Some("element1 documentation"),
      symbol_range: "...",
      full_range: "...",
    ),
    EnumElement(
      name: "ELEMENT2",
      value: Some("\"quattro\""),
      doc: None,
      symbol_range: "...",
      full_range: "...",
    ),
    EnumElement(
      name: "ELEMENT3",
      value: None,
      doc: None,
      symbol_range: "...",
      full_range: "...",
    ),
  ],
  annotations: [],
  doc: None,
  full_range: "...",
  symbol_range: "...",
)
